functions/security-requirements.js

```diff
@@ -0,0 +1,73 @@
+// Check:
+// - each entry of a global or operation `security` property references a defined
+//   security scheme.
+// - all scopes referenced by an "oauth2" entry are defined in the corresponding
+//   security scheme.
+
+// @param input - a security property (global or operation)
+module.exports = (input, options, context) => {
+  if (input === null || !Array.isArray(input)) {
+    return [];
+  }
+
+  const isObject = (obj) => obj && typeof obj === 'object';
+  const oas2Schemes = (doc) => (isObject(doc.securityDefinitions) ? doc.securityDefinitions : {});
+  const oas3Schemes = (doc) => (isObject(doc.components) && isObject(doc.components.securitySchemes)
+    ? doc.components.securitySchemes : {});
+
+  const oasDoc = context.document.data;
+  const schemes = oasDoc.swagger ? oas2Schemes(oasDoc) : oas3Schemes(oasDoc);
+
+  const path = context.path || [];
+
+  const errors = [];
+  input.forEach((securityReq, index) => {
+    // oas2-schema requires securityReq to be an object.
+    // Checking here just to avoid runtime errors.
+    if (isObject(securityReq)) {
+      Object.keys(securityReq).forEach((key) => {
+        if (!schemes[key]) {
+          errors.push({
+            message: `Security scheme "${key}" is not defined.`,
+            path: [...path, index, key],
+          });
+          return;
+        }
+
+        const scheme = schemes[key];
+        // oas2-schema requires scheme to be an object.
+        // Checking here just to avoid runtime errors.
+        if (!isObject(scheme)) { return; }
```

Comment:
I think LN40 will be tested by [the test for malformed API doc](https://github.com/Azure/azure-api-style-guide/pull/82/files#diff-2e4fcd01a70a3982c625cb45fff6916518662e9b8b887d34a945748eeaf47eceR249), because the "oauth2" security scheme is not an object.